src/seedsigner/views/seed_views.py

```diff
@@ -2294,3 +2301,289 @@ def run(self):
 
         # Exiting/Canceling the QR display screen always returns Home
         return Destination(MainMenuView, skip_current_view=True)
+
+
+
+"""****************************************************************************
+    Shamir's Secret Sharing Views
+****************************************************************************"""
+class SeedEntryShamirThresholdView(View):
+    def __init__(self):
+        super().__init__()
+
+
+    def run(self):
+        title = _("SLIP-39 Threshold")
+        ret_dict = self.run_screen(seed_screens.SeedEntryShamirThresholdScreen, entered_number="", title=title)
+
+        if "is_back_button" in ret_dict:
+            return Destination(BackStackView)
+            
+        elif ret_dict["entered_number"] != "" and ret_dict["entered_number"] != "0":
+            return Destination(SeedShamirShareImportSelectWordCount, view_args={"k": int(ret_dict["entered_number"])})
+
+        else:
+            if ret_dict["entered_number"] == "0":
+                from seedsigner.gui.toast import ErrorToast
+                self.controller.activate_toast(ErrorToast(_("Threshold not valid")))
+
+        return Destination(BackStackView)
+        
+
+
+class SeedShamirShareImportSelectWordCount(View):
+    TYPE_12WORD = ButtonOption("20 words (128-bit entropy)")
+    TYPE_24WORD = ButtonOption("33 words (256-bit entropy)")
+
+    def __init__(self, k: int):
+        super().__init__()
+        self.k = k
+
+
+    def run(self):
+        button_data = [self.TYPE_12WORD, self.TYPE_24WORD]
+
+        selected_menu_num = self.run_screen(
+            ButtonListScreen,
+            title=_("Select SLIP-39 Word Count"),
+            button_data=button_data,
+        )
+        
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(BackStackView)
+
+        elif button_data[selected_menu_num] == self.TYPE_12WORD:
+            self.controller.storage.init_pending_shamir_share_set(num_words=20, num_shares=self.k)
+            return Destination(SeedShamirShareMnemonicEntryView)
+        
+        elif button_data[selected_menu_num] == self.TYPE_24WORD:
+            self.controller.storage.init_pending_shamir_share_set(num_words=33, num_shares=self.k)
+            return Destination(SeedShamirShareMnemonicEntryView)
+
+
+
+class SeedShamirShareMnemonicEntryView(View):
+    def __init__(self, cur_word_index: int = 0, is_calc_final_word: bool=False, cur_set_index: int = 0):
+        super().__init__()
+        self.cur_word_index = cur_word_index
+        self.cur_word = self.controller.storage.get_pending_mnemonic_word(cur_word_index)
+        self.is_calc_final_word = is_calc_final_word
+        self.cur_set_index = cur_set_index
+
+
+    def run(self):
+        ret = self.run_screen(
+            seed_screens.SeedMnemonicEntryScreen,
```

Comment:
I ran some sanity checks on the `SLIP39_WORDS` list to make sure the right side word selection column would still be properly spaced.

The longest words in the list are 8 chars, same as bip39 ("mushroom"). And they're rendered with a fixed width font so spacing is good.